src/anemoi/training/train/forecaster.py

```diff
@@ -359,6 +380,10 @@ def rollout_step(
         # for validation not normalized in-place because remappers cannot be applied in-place
         batch = self.model.pre_processors(batch, in_place=not validation_mode)
 
+        if not self.updated_loss_mask:
+            # update loss scalar after first application and initialization of preprocessors
+            self.training_weights_for_imputed_variables(batch)
+
```

Comment:
What if a user has set the number of sanity_steps to 0, `validation` will not run before training, and thus the mask won't be set

Reply:
The mask is set in the first application of the preprocessors, which are applied in both training and validation mode.

Reply:
But the scalar won't be set.

Reply:
Isn't the scalar of the loss function set in the initialization of the model?

Reply:
Sorry, the scalar will be set in init to `1`.
But, if the validation_step is not run before training, the scalar will not be updated with the true mask value.